### PR TITLE
CI update

### LINKED ISSUE #5: Set up lottery reviewing
Use [this GitHub Action](https://github.com/marketplace/actions/reviewer-lottery). Create an escape route to skip code review when a feature was pair-programmed.

**Acceptance criteria**:
- every subsequent PR should have 1 person from the team assigned for review randomly **after the CI checks pass**
- PRs with a `pair-programmed` label should not be affected.

### LINKED ISSUE #6: Set up a CI checking workflow
Blocked by #9 and #10.

**Acceptance criteria**:
- every PR is validated with `pnpm run lint` in the workspace root and `pnpm run test:unit` and `pnpm run test:e2e` in the `app` project
- pnpm's dependencies are cached across workflow runs

### PR DESCRIPTION
**Pull request by alkaitagi**
Added lottery review and CI linting with caching. Closes #5, closes #6.

### CODE REVIEW

File: .github/workflows/unit-test-pr.yml [hunk 1]
```diff
@@ -16,7 +16,15 @@ jobs:
         with:
           version: 6.28.0
 
+      - name: Cache 📦
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+          cache: 'pnpm'
```
Comment (illright) [inline .github/workflows/unit-test-pr.yml:23]: praise: wow, I must have been asleep for too long. This is way awesome. I'm used to manually setting up caches by seeking out third-party actions :)

File: .github/workflows/auto-reviewer.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+name: "Reviewer lottery"
+on:
+  pull_request:
+    types: [opened, ready_for_review, reopened]
+    tags-ignore:
+    - pair-programmed
```
Comment (illright) [inline .github/workflows/auto-reviewer.yml:6]: nitpick (non-blocking): This looks slightly misindented, I imagine Prettier will want to reformat it

File: .github/workflows/auto-reviewer.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
```
Comment (illright) [inline .github/workflows/auto-reviewer.yml:12]: question: is a checkout even needed here?

**Review by illright — APPROVED** [review]

File: .github/workflows/unit-test-pr.yml (hunk 1 above)
Comment (illright) [reply .github/workflows/unit-test-pr.yml:23]: Yo @aabounegm check this out

File: .github/workflows/auto-reviewer.yml (hunk 3 above)
Comment (illright) [reply .github/workflows/auto-reviewer.yml:12]: Never mind, I understood that it's for the config of the lottery review

File: .github/workflows/unit-test-pr.yml (hunk 1 above)
Comment (aabounegm) [reply .github/workflows/unit-test-pr.yml:23]: Yeah, I noticed recently that this action now handles caching as well, but didn't realize that it did for other package managers as well)

**Pull request merged**